plugin/init.lua

```diff
@@ -108,37 +114,17 @@ local function execute_cmd_with_stdin(cmd, input)
 	end
 end
 
----@alias encryption_opts {enable: boolean, private_key: string | nil, public_key: string | nil, encrypt: fun(file_path: string, lines: string), decrypt: fun(file_path: string): string | nil}
+---@alias encryption_opts {enable: boolean, private_key: string | nil, public_key: string | nil, encrypt: fun(file_path: string, lines: string): (boolean, string), decrypt: fun(file_path: string): (boolean, string, string)}
 pub.encryption = {
 	enable = false,
 	private_key = nil,
 	public_key = nil,
 	encrypt = function(file_path, lines)
-		wezterm.emit("resurrect.encrypt.start", file_path)
 		local cmd = string.format("age -r %s -o %s", pub.encryption.public_key, file_path:gsub(" ", "\\ "))
-		local success, output = execute_cmd_with_stdin(cmd, lines)
-
-		if not success then
-			wezterm.emit("resurrect.error", "Encryption failed: " .. output)
-			wezterm.log_error("Encryption failed: " .. output)
-			return
-		end
-		wezterm.emit("resurrect.encrypt.finished", file_path)
+		return execute_cmd_with_stdin(cmd, lines)
```

Comment:
Sorry if I was unclear in my previous change request.

What I meant was using the buildin `error` function when the command is unsuccessful, then in the `write_state` a `pcall` like so:
```lua
-- other code
	encrypt = function(file_path, lines)
		local cmd = string.format("age -r %s -o %s", pub.encryption.public_key, file_path:gsub(" ", "\\ "))
		local success, output = execute_cmd_with_stdin(cmd, lines)
		if not success then
		  error("Encryption failed:" .. output)
		end
	end
	-- other code ...
	local function write_state(file_path, state)
	wezterm.emit("resurrect.save_state.start", file_path)
	local json_state = wezterm.json_encode(state)
	json_state = sanitize_json(json_state)
	if pub.encryption.enable then
		wezterm.emit("resurrect.encrypt.start", file_path)
		local ok, err = pcall(function() return pub.encryption.encrypt(file_path, json_state)end)
		If not ok then
		  wezterm.emit("resurrect.error", err)
			wezterm.log_error(err)
		else
			wezterm.emit("resurrect.encrypt.finished", file_path)
		end
	else
		-- ...
end
	
	```

The same for decrypt:
```lua
-- other code
decrypt = function(file_path)
		local success, stdout, stderr =
			wezterm.run_child_process({ "age", "-d", "-i", pub.encryption.private_key, file_path })
		if not success then
		  error("Decryption failed: " .. stderr)
		end
		return stdout
	end
	-- other code...
local function load_json(file_path)
	local json
	if pub.encryption.enable then
		wezterm.emit("resurrect.decrypt.start", file_path)
    local ok, output = pcall(function() return pub.encryption.decrypt(file_path) end)
		if not ok then
			wezterm.emit("resurrect.error", "Decryption failed: " .. output)
			wezterm.log_error("Decryption failed: " .. output)
		else
			if is_windows then
				json = output:gsub('`"', '"'):gsub("`n", "\n"):gsub("`r", "\r")
			end
			wezterm.emit("resurrect.decrypt.finished", file_path)
		end
	else
		local lines = {}
		for line in io.lines(file_path) do
			table.insert(lines, line)
		end
		json = table.concat(lines)
	end
	if not json then
		return nil
	end
	json = sanitize_json(json)
	return wezterm.json_parse(json)
end
```

This way if a user defined de/encrypt function errors, then it is still possible to handle the error like we would with the buildin functions

Reply:
Sorry I misunderstood. I'll implement those changes. 

With the `pcall`, since the function will continue safely instead of erroring and failing, I needed to add an early return in `workspace_state.lua` to deal with this scenario because `workspace_state` would be nil.

```lua
function pub.restore_workspace(workspace_state, opts)
	wezterm.emit("resurrect.workspace_state.restore_workspace.start")
	if workspace_state == nil then
		return
	end
```